### DISCUSSION TITLE
MPI path missing?

### DISCUSSION THREAD

**Discussion by Aaboe-KU** [OpenRadioss/OpenRadioss#3217] [Q&A]
I'm having issues with getting MPI up and running in the newest version of the OpenRadioss GUI. Within *OpenRadioss_gui.py*, the following function is 

```python
def check_mpi_path(self):
     mpi_path_file = "path_to_intel-mpi.txt"
     if np_entry.get_user_input() > '1' and not os.path.exists(mpi_path_file):
         messagebox.showinfo('', 'Running MPI requires intel mpi installation. Please browse to an intel-mpi location. Or select np = 1')
         directory_path = filedialog.askdirectory(
             title='Select intel-mpi directory'            
         )
         if directory_path:
             with open(mpi_path_file, 'w') as file:
                 file.write('"' + directory_path + '"')
     else:
         # MPI path file exists or np <= 1, continue with the script
         pass
``` 

Running the GUI with np $\geq$ 1, I do not get prompted to browse to an intel-mpi location. All I am prompted with is: *'mpiexec' is not recognized as an internal or external command, operable program or batch file.*

I've ensured that oneAPI is properly installed, and I've even attempted to add mpiexec directly to PATH. I've attempted to create the "path_to_intel-mpi.txt" and added the path to the vars.bat within the MPI environment. I also tried adding the path to the binaries and even mpiexec itself. None of these worked.

Is there an issue with how this path is handled? Any ideas?

**Comment by PaulAltair** [comment]
Yes, this is known and there is a ticket for it. see [OpenRadioss/Tools/Issues/25](https://github.com/OpenRadioss/Tools/issues/25)

With the latest release, we temporarily removed the 'request path for MPI' subroutine, (the function code you see there is no longer called), we plan to implement a different simpler way again soon. 

For now, you can add to path envvar: the following paths: and mpi runs should function again (_mpilocation_ is your intel mpi install location)

_mpilocation_/libfabric/bin
_mpilocation_/bin
_mpilocation_/bin/release
_mpilocation_/bin/utils

**Comment by OlivierWienholtz** [comment]
Hello, 
I recommend to call the IntelMPI vars.bat to set all need environemnts for IntelMPI prior invoking openradioss_gui  :

call [Path to OneAPI]\mpi\latest\env\vars.bat
(On my installation : call "c:\Program Files (x86)\Intel\oneAPI\mpi\latest\env\vars.bat")

We will add some settings in the future.

**Reply by bernoertel** [reply]
Hello Mr. Altair & Wienholz, I use OpenRadioss under Win10.  Till now I had no MPI  installed (--> slow) and have now installed  Microsoft MPI. I assume, that is wrong ( I have no vars-file) and  above stands IntelMPI is to use.  I have nothing found in the Open Radioss Docu, but need instructions,  what I must install and how to configure, so that I have/get a vars-file for OpenRadioss.

**Reply by MarianBulla** [reply]
Hello @bernoertel ,
you can find a detailed docu on how to get OpenRadioss installed (with the Intel compilers and Inte MPI) here: 
https://github.com/OpenRadioss/OpenRadioss/blob/main/HOWTO.md#build-environment-on-windows

You can also review on YouTube, how it can be done with a step-by-step demonstration of the installation procedure:
https://www.youtube.com/@OpenRadiossCommunity

Hope it helps you ?

Thanks and best regards,
Marian

**Reply by bernoertel** [reply]
Hello Hr. Bulla, sorry when I ask further :  is it really mandatory to install Open Radioss from the sources, when I want to use it together with MPI  ??  I have installed  from the benaries a (slow) running version without MPI and so I get this error :
![image](https://github.com/user-attachments/assets/5cc60ee5-1acf-4b58-a934-f6b4a91a9efc)
To add a MPI  ( IntelMPI ? )  subsequently is impossible  ??   Greetings   bernoertel

**Reply by MarianBulla** [reply]
Hello,
no, you don't need the sources. You can use the precompiled executables.
But you need the intel mpi libraries.

Hope it helps you ?

Thanks and best
Marian

**Comment by OlivierWienholtz** [comment]
Hello @bernoertel 

It is true, that Microsoft MPI will not work with OpenRadioss.

In Addition to Marian's post:
If you are downloading the OpenRadioss releases, it is true that you don't need the full OneAPI package.
Intel MPI can be downloaded as a separate package here :
https://www.intel.com/content/www/us/en/developer/tools/oneapi/mpi-library-download.html

Best Regards,
Olivier

**Reply by bernoertel** [reply]
Hello  Mr. Wienholz
my Prozessor  :  Intel(R) Xeon(R) CPU E5-2690 0 @ 2.90GHz, 2901 MHz, 
                          8 Kern(e), 16 logische(r) Prozessor(en)

I have installed INTEL-MPI in this PATH :
C:\Program Files (x86)\Intel\oneAPI\mpi\2021.14
When Open Radioss (OR) asked me (when np=2) 
I gave OR the vars.bat from 
C:\Program Files (x86)\Intel\oneAPI\mpi\2021.14\env\vars.bat
and it never asked again (stored the path??)
Now I have the following performance :

-nt	-np	time  %CPU
------------------------------
1	1	61s	12
8	1	22s	80
8	2	27s	100
16	2	46s	100
1	8	20s	70   (**)
2	8	40s	100
------------------------------
(**) is the fastest run.
Can you see, if INTEL-MPI is used or not ?
Sincerly    bernoertel

**Reply by OlivierWienholtz** [reply]
Hello @bernoertel,

In the meanings : 
-nt is number of threads
-np is number of MPI processes.

To run with -np, you need IntelMPI running.

This information is also written in the header of the OpenRadioss Output file ([Deckname]_0001.out)
SPMD Domains = MPI processes
Threads per domain = Threads..

 NEON1M11_0001.out
************************************************************************
**                                                                    **
**                                                                    **
**                         OpenRadioss Engine                         **
**                                                                    **
**            Non-linear Finite Element Analysis Software             **
**                                                                    **
**                                                                    **
**                                                                    **
**             Windows 64 bits, Intel compiler, Intel MPI             **
**                      Double Precision Version                      **
**                                                                    **
**                                                                    **
**                                                                    **
************************************************************************
** OpenRadioss Software                                               **
** COPYRIGHT (C) 1986-2025 Altair Engineering, Inc.                   **
** Licensed under GNU Affero General Public License.                  **
** See License file.                                                  **
************************************************************************
 
 **NUMBER OF SPMD DOMAINS           4
 NUMBER OF THREADS PER DOMAIN     1
 NUMBER OF HMPP PROCESSES         4**
  
 COMPUTATION HARDWARE DESCRIPTION


Based on your table, If I read it correctly 
1  8 20s70 
is 1 Thread, 8 MPI processes, took 20s 70
So yes, this job used MPI. You IntelMPI installation was found & used, otherwise it would not work.

Best Regards,
Olivier Wienholtz

**Comment by OlivierWienholtz** [comment]
Hello @bernoertel

You need to have the at least the IntelMPI package or the Full OneAPI package installed.

We have a known issue : 
The launch script does not set the Intel MPI variables, they must be loaded before invoking the script.
I have seen that sometimes the Windows system has those variables set, so could be used by any program, and sometimes not.
If not MPI is not found & launch script may not work.
You can find the description here : https://github.com/OpenRadioss/Tools/issues

In this case, I would suggest to use a shell which has the variables & invoke the GUI or create a .bat file which does both. 

Hope it helps,
Olivier
